Core/src/org/sleuthkit/autopsy/centralrepository/AddEditCentralRepoCommentAction.java

```diff
@@ -0,0 +1,144 @@
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle.Messages;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.centralrepository.datamodel.CorrelationAttribute;
+import org.sleuthkit.autopsy.centralrepository.datamodel.CorrelationCase;
+import org.sleuthkit.autopsy.centralrepository.datamodel.CorrelationDataSource;
+import org.sleuthkit.autopsy.centralrepository.datamodel.EamArtifactUtil;
+import org.sleuthkit.autopsy.centralrepository.datamodel.EamDb;
+import org.sleuthkit.autopsy.centralrepository.datamodel.EamDbException;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * An AbstractAction to manage adding and modifying a Central Repository file
+ * instance comment.
+ */
+public final class AddEditCentralRepoCommentAction extends AbstractAction {
+
+    private static final Logger logger = Logger.getLogger(AddEditCentralRepoCommentAction.class.getName());
+    
+    private boolean addToDatabase;
+    private CorrelationAttribute correlationAttribute;
+
+    /**
+     * Private constructor to create an instance given a CorrelationAttribute.
+     *
+     * @param correlationAttribute The correlation attribute to modify.
+     * @param displayName          The text for the menu item.
+     */
+    private AddEditCentralRepoCommentAction(CorrelationAttribute correlationAttribute, String displayName) {
+        super(displayName);
+        this.correlationAttribute = correlationAttribute;
+    }
+
+    /**
+     * Private constructor to create an instance given an AbstractFile.
+     *
+     * @param file        The file from which a correlation attribute to modify
+     *                    is derived.
+     * @param displayName The text for the menu item.
+     */
+    private AddEditCentralRepoCommentAction(AbstractFile file, String displayName) throws EamDbException, NoCurrentCaseException, TskCoreException {
+
+        super(displayName);
+        correlationAttribute = EamArtifactUtil.getCorrelationAttributeFromContent(file);
+        if (correlationAttribute == null) {
+            addToDatabase = true;
+            correlationAttribute = EamArtifactUtil.makeCorrelationAttributeFromContent(file);
+        }
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent event) {
+        addEditCentralRepoComment();
+    }
+
+    /**
+     * Create a Add/Edit dialog for the correlation attribute file instance
+     * comment. The comment will be updated in the database if the file instance
+     * exists there, or a new file instance will be added to the database with
+     * the comment attached otherwise.
+     */
+    public void addEditCentralRepoComment() {
+        CentralRepoCommentDialog centralRepoCommentDialog = new CentralRepoCommentDialog(correlationAttribute);
+        centralRepoCommentDialog.display();
```

Comment:
I think the dialog would look nicer with a title. Our logo would be nice, too, if it's not too much work.